measures/TableMeasures/ScalarQuantColumn.tcc

```diff
@@ -234,7 +234,42 @@ void ScalarQuantColumn<T>::put (uInt rownr, const Quantum<T>& q)
   }
 }
 
-} //# NAMESPACE CASACORE - END
+template<class T>
+SHARED_PTR<Quantum<Vector<T> > > ScalarQuantColumn<T>::getColumn(const Unit* unit) const {
+    SHARED_PTR<Quantum<Vector<T> > > qv;
+    const Unit* unitOut = unit ? unit
+        : itsConvOut ? &itsUnitOut : NULL;
+    if (itsUnitsCol) {
+        Vector<String> units = itsUnitsCol->getColumn();
+        Vector<String>::const_iterator uiter = units.begin();
+        Unit first(*uiter);
+        if (! unitOut) {
+            unitOut = &first;
+        }
+        qv.reset(new Quantum<Vector<T> >(Vector<T>(), *unitOut));
+        Vector<T>& val = qv->getValue();
+        itsDataCol->getColumn(val);
+        typename Vector<T>::iterator viter = val.begin();
+        typename Vector<T>::iterator vend = val.end();
+        Quantum<T> q;
+        for (; viter != vend; ++viter, ++uiter) {
+            q.setValue(*viter);
+            q.setUnit(*uiter);
+            *viter = q.getValue(*unitOut);
+        }
+    }
+    else {
+        qv.reset(new Quantum<Vector<T> >(Vector<T>(), itsUnit));
+        Vector<T>& val = qv->getValue();
+        itsDataCol->getColumn(val);
+        if (unitOut) {
+            val = qv->getValue(*unitOut);
+            qv->setUnit(*unitOut);
+        }
```

Comment:
just though it would be more consistent, but thats slower and unit identity comparison looks clumsy so this is fine too

though couldn't it just be `qv->convert(unitOut)`?